integration/ztoc_test.go

```diff
@@ -97,3 +116,214 @@ func TestSociZtocList(t *testing.T) {
+					t.Fatal("failed to return error")
+				}
+			})
+		}
+	}
+
+}
+
+func TestZtocGetFile(t *testing.T) {
+	t.Parallel()
+	sh, done := newSnapshotterBaseShell(t)
+	defer done()
+	rebootContainerd(t, sh, "", "")
+
+	testImages := prepareSociIndices(t, sh)
+
+	var (
+		tempOutputStream = "test.txt"
+		randomFile       = "ab2423n2j3h2jh32jn323"
+		randomZtocDigest = testutil.GenerateRandomDigest()
+	)
+
+	getFilePathWithinZtoc := func(t *testing.T, ztocDigest string) string {
+		var (
+			zinfo Info
+			path  string
+		)
+		output := sh.O("soci", "ztoc", "info", ztocDigest)
+		json.Unmarshal(output, &zinfo)
+		for _, file := range zinfo.Files {
+			if file.Size > 0 {
+				path = file.Filename
+			}
+		}
+		return path
+	}
+	verifyOutputStream := func(t *testing.T, sh *dockershell.Shell, contents, output []byte) {
+		if len(output) > 0 {
+			output = output[:len(output)-1]
+			res := bytes.Compare(contents, output)
+			if res != 0 {
+				t.Fatalf("output: %s expected: %s", output, contents)
+			}
+
+		} else {
+			o := sh.O("cat", tempOutputStream)
+			res := bytes.Compare(contents, o)
+			if res != 0 {
+				t.Fatalf("output: %s expected: %s", o, contents)
+			}
+
+		}
+	}
+
+	for _, img := range testImages {
+		img := img
+		tempLayerPath := fmt.Sprintf("layer-%s", img.imgName)
+		ztocDigest := img.ztocDigests[0]
+		fileName := getFilePathWithinZtoc(t, ztocDigest)
+
+		testCases := []struct {
+			name        string
+			cmd         []string
+			expectedErr bool
+		}{
+			{
+				name:        "Ztoc that does not exist",
+				cmd:         []string{"soci", "ztoc", "get-file", randomZtocDigest, randomFile},
+				expectedErr: true,
+			},
+			{
+				name:        "Ztoc exists but file does not exist",
+				cmd:         []string{"soci", "ztoc", "get-file", ztocDigest, randomFile},
+				expectedErr: true,
+			},
+			{
+				name:        "Ztoc and file exists, file contents redirected to stdout",
+				cmd:         []string{"soci", "ztoc", "get-file", ztocDigest, fileName},
+				expectedErr: false,
+			},
+			{
+				name:        "Ztoc and file exists, file contents redirected to output file",
+				cmd:         []string{"soci", "ztoc", "get-file", "-o", tempOutputStream, ztocDigest, fileName},
+				expectedErr: false,
+			},
+		}
+		for _, tt := range testCases {
+			t.Run(tt.name, func(t *testing.T) {
+				output, err := sh.OLog(tt.cmd...)
+				if !tt.expectedErr {
+					if err != nil {
+						t.Fatalf("failed to return file contents")
+					}
+					var layerDigest string
+					sociIndex, _ := sociIndexFromDigest(sh, img.sociIndexDigest)
+
+					for _, blob := range sociIndex.Blobs {
+						if blob.Digest.String() == ztocDigest {
+							layerDigest = blob.Annotations[soci.IndexAnnotationImageLayerDigest]
+							break
+						}
+					}
+
+					layerPath := filepath.Join(containerdBlobStorePath, trimSha256Prefix(layerDigest))
+					contents := sh.X("mkdir", "-p", tempLayerPath).
+						X("tar", "-xzf", layerPath, "-C", tempLayerPath).
```

Comment:
Instead of doing this, can we just use `gzip.NewReader` with `tar.NewReader`? 

For example, something like:

```
layerContents := sh.O("cat", layerPath)
gzipReader := gzip.NewReader(bytes.NewReader(layerContents))
tarReader := tar.NewReader(gzipReader)

// for each file in tarReader, if regular file, compare contents
```